src/App.tsx

```diff
@@ -1,24 +1,280 @@
+          setErrorText('');
+        } else {
+          setErrorText('Unable to update todo');
+        }
+
+        closeError();
+      } finally {
+        setLoadingTodoId([]);
+      }
+    }, [todos],
+  );
+
+  const toggleAllCompleted = async () => {
+    try {
+      const activeIds = activeTodos.map(activeTodo => activeTodo.id);
+      const completedIds = completedTodos.map(
+        completedTodo => completedTodo.id,
+      );
+
+      let toggleTodos;
+      let updatedTodos;
+
+      if (activeTodos.length) {
+        setLoadingTodoId(activeIds);
+
+        toggleTodos = activeTodos.map(active => (
+          patchTodo(active.id, { ...active, completed: true })
+        ));
+
+        updatedTodos = todos.map(todo => (
+          activeIds.includes(todo.id) ? { ...todo, completed: true } : todo
+        ));
+      } else {
+        setLoadingTodoId(completedIds);
+
+        toggleTodos = completedTodos.map(completedTodo => (
+          patchTodo(completedTodo.id, { ...completedTodo, completed: false })
+        ));
+
+        updatedTodos = todos.map(todo => (
+          completedIds.includes(todo.id)
+            ? { ...todo, completed: false } : todo
+        ));
+      }
+
+      await Promise.all(toggleTodos);
+      setTodos(updatedTodos);
+    } catch {
+      setErrorText('Unable to toggle all todos');
+    } finally {
+      setLoadingTodoId([]);
+    }
+  };
+
+  const removeCompletedTodos = async () => {
+    try {
+      const completeIds = completedTodos.map(({ id }) => removeTodo(id));
+
+      await Promise.all(completeIds);
+      setTodos(activeTodos);
+    } catch {
+      setErrorText('Unable to delete completed todos');
+      closeError();
+    }
+  };
+
+  const handleSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (query.trim().length) {
+      addTodo(query);
+    } else {
+      setErrorText('Title is empty');
+      closeError();
+    }
+
+    setQuery('');
+  };
+
+  const visibleTodos = useMemo(() => {
+    switch (filterType) {
+      case FilterType.Completed:
+        return todos.filter(todo => todo.completed);
+
+      case FilterType.Active:
+        return todos.filter(todo => !todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [todos, filterType]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
```

Comment:
fix this linter error or disable linter for this line or change linter rule itself. But try not to push code with some errors